zoo_frontend/components/VirtualTable.jsx

```diff
@@ -0,0 +1,235 @@
+        align={columns[columnIndex].numeric || false ? 'right' : 'left'}
+      >
+        {inner}
+      </TableCell>
+    );
+  };
+
+  render() {
+    const { classes, columns, ...tableProps } = this.props;
+    return (
+      <AutoSizer>
+        {({ height, width }) => (
+          <Table
+            className={classes.table}
+            height={height}
+            width={width}
+            {...tableProps}
+            rowClassName={this.getRowClassName}
+          >
+            {columns.map(({
+              cellContentRenderer = null, className, dataKey, ...other
+            }, index) => {
+              let renderer;
+              if (cellContentRenderer != null) {
+                renderer = cellRendererProps => this.cellRenderer({
+                  cellData: cellContentRenderer(cellRendererProps),
+                  columnIndex: index,
+                });
+              } else {
+                renderer = this.cellRenderer;
+              }
+
+              return (
+                <Column
+                  key={dataKey}
+                  headerRenderer={headerProps => this.headerRenderer({
+                    ...headerProps,
+                    columnIndex: index,
+                  })
+                  }
+                  className={classNames(classes.flexContainer, className)}
+                  cellRenderer={renderer}
+                  dataKey={dataKey}
+                  {...other}
+                />
+              );
+            })}
+          </Table>
+        )}
+      </AutoSizer>
+    );
+  }
+}
+
+MuiVirtualizedTable.propTypes = {
+  classes: PropTypes.object.isRequired,
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      cellContentRenderer: PropTypes.func,
+      dataKey: PropTypes.string.isRequired,
+      width: PropTypes.number.isRequired,
+    }),
+  ).isRequired,
+  headerHeight: PropTypes.number,
+  onRowClick: PropTypes.func,
+  rowClassName: PropTypes.string,
+  rowHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.func]),
+  sort: PropTypes.func,
+};
+
+MuiVirtualizedTable.defaultProps = {
+  headerHeight: 56,
+  rowHeight: 56,
+};
+
+const WrappedVirtualizedTable = withStyles(styles)(MuiVirtualizedTable);
+
+const data = [
+  ['Cucumber', '1500g', 'Sea Turtle', 899, 'Aquarium'],
+  ['Apples', '2000g', 'Gorilla', 765, 'Jungle'],
+];
+
+let id = 0;
+function createData(dessert, calories, fat, carbs, protein) {
+  id += 1;
+  return {
+    id, dessert, calories, fat, carbs, protein,
+  };
+}
+
+const rows = [];
+
+for (let i = 0; i < 100; i += 1) {
+  const randomSelection = data[Math.floor(Math.random() * data.length)];
+  rows.push(createData(...randomSelection));
+}
+
+function ReactVirtualizedTable() {
+  return (
+    <Paper style={{ height: 500, width: '100%' }}>
+      <WrappedVirtualizedTable
+        rowCount={rows.length}
+        rowGetter={({ index }) => rows[index]}
+        onRowClick={event => console.log(event)}
+        columns={[
```

Comment:
columns should come in as props, rowCount and rowGetter can be figured out right above this return based off the props send in @ line 191.